connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
Usually the implementations of `stop()` that I've seen are asynchronous, so we might not get much benefit from parallel task shutdown either. However, the javadoc does not say that `stop()` must be asynchronous, so it might still be worthwhile.